ui/src/app/components/breadcrumb.tsx

```diff
@@ -111,42 +142,60 @@ const BreadcrumbLink = ({href, ...props}) => {
   />;
 };
 
-const Breadcrumb = fp.flow(
+interface Props {
+  workspace: WorkspaceData;
+  cohort: Cohort;
+  conceptSet: ConceptSet;
+  urlParams: any;
+  routeConfigData: any;
+}
+
+export const Breadcrumb = fp.flow(
   withCurrentWorkspace(),
   withCurrentCohort(),
   withCurrentConceptSet(),
   withUrlParams(),
   withRouteConfigData()
-)(({workspace, cohort, conceptSet, urlParams, routeConfigData}) => {
-  const trail = getTrail(
-    routeConfigData.breadcrumb,
-    {urlParams, workspace, conceptSet, cohort}
-  );
-  const first = fp.dropRight(1, trail);
-  const last = fp.last(trail);
-  return <div style={{marginLeft: '3.25rem', display: 'inline-block'}}>
-    {first.map(({label, url}, i) => {
-      return <React.Fragment key={i}>
-        <BreadcrumbLink href={url} style={styles.firstLink}>
-          {label}
-        </BreadcrumbLink>
-        <span style={{color: colors.primary}}> &gt; </span>
-      </React.Fragment>;
-    })}
-    {last && <div>
-      <BreadcrumbLink href={last.url} style={styles.lastLink}>
-        {last.label}
-      </BreadcrumbLink>
-    </div>}
-  </div>;
-});
+)(
+  class extends React.Component<Props> {
+    constructor(props) {
+      super(props);
+    }
+
+    trail(): Array<BreadcrumbData> {
+      return getTrail(
+        this.props.routeConfigData.breadcrumb,
+        this.props.workspace,
+        this.props.cohort,
+        this.props.conceptSet,
+        this.props.urlParams
+      );
+    }
 
-@Component({
-  selector: 'app-breadcrumb',
-  template: '<div #root></div>'
-})
-export class BreadcrumbComponent extends ReactWrapperBase {
-  constructor() {
-    super(Breadcrumb, []);
+    first(): Array<BreadcrumbData> {
+      return fp.dropRight(1, this.trail());
```

Comment:
i like this, it would be nice to have it in a util file that takes in an array and returns first last. (right now its index.tsx file that has other array functions maybe we can add to that or create a new util file)